### PR TITLE
Most of NIFI-9096 through NIFI-9142 (JUnit 5)

### PR DESCRIPTION
**Pull request by MikeThomsen**
Thank you for submitting a contribution to Apache NiFi.

Please provide a short description of the PR here:

#### Description of PR

_Enables X functionality; fixes bug NIFI-YYYY._

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [ ] Is there a JIRA ticket associated with this PR? Is it referenced 
     in the commit message?

- [ ] Does your PR title start with **NIFI-XXXX** where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.

- [ ] Has your PR been rebased against the latest commit within the target branch (typically `main`)?

- [ ] Is your initial contribution a single, squashed commit? _Additional commits in response to PR reviewer feedback should be made on this branch and pushed to allow change tracking. Do not `squash` or use `--force` when pushing to allow for clean monitoring of changes._

### For code changes:
- [ ] Have you ensured that the full suite of tests is executed via `mvn -Pcontrib-check clean install` at the root `nifi` folder?
- [ ] Have you written or updated unit tests to verify your changes?
- [ ] Have you verified that the full build is successful on JDK 8?
- [ ] Have you verified that the full build is successful on JDK 11?
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)? 
- [ ] If applicable, have you updated the `LICENSE` file, including the main `LICENSE` file under `nifi-assembly`?
- [ ] If applicable, have you updated the `NOTICE` file, including the main `NOTICE` file found under `nifi-assembly`?
- [ ] If adding new Properties, have you added `.displayName` in addition to .name (programmatic access) for each of the new properties?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered?

### Note:
Please ensure that once the PR is submitted, you check GitHub Actions CI for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

**Comment by MikeThomsen** [comment]
@exceptionfactory cleaned up and ready to go.

File: nifi-nar-bundles/nifi-gcp-bundle/nifi-gcp-processors/src/test/java/org/apache/nifi/processors/gcp/storage/PutGCSObjectTest.java [hunk 1]
```diff
@@ -48,25 +68,6 @@
...
-import org.apache.nifi.flowfile.attributes.CoreAttributes;
-import org.apache.nifi.util.MockFlowFile;
-import org.apache.nifi.util.TestRunner;
-import org.junit.Test;
-import org.mockito.ArgumentCaptor;
-import org.mockito.Captor;
-import org.mockito.Mock;
-
 /**
  * Unit tests for {@link PutGCSObject} which do not use Google Cloud resources.
```
Comment (exceptionfactory) [inline nifi-nar-bundles/nifi-gcp-bundle/nifi-gcp-processors/src/test/java/org/apache/nifi/processors/gcp/storage/PutGCSObjectTest.java:72]: It looks like this class is still using JUnit 4 assertions.

File: nifi-nar-bundles/nifi-hadoop-bundle/nifi-hdfs-processors/src/test/java/org/apache/nifi/processors/hadoop/AbstractHadoopTest.java [hunk 2]
```diff
@@ -117,26 +117,35 @@ public void testErrorConditions() {
...
+//        try {
+//            final File brokenCoreSite = new File("src/test/resources/core-site-broken.xml");
+//            final ResourceReference brokenCoreSiteReference = new FileResourceReference(brokenCoreSite);
+//            final ResourceReferences references = new StandardResourceReferences(Collections.singletonList(brokenCoreSiteReference));
+//            final List<String> locations = references.asLocations();
+//            processor.resetHDFSResources(locations, runner.getProcessContext());
+//            Assert.fail("Should have thrown SocketTimeoutException");
+//        } catch (IOException e) {
+//            e.printStackTrace();
+//        }
```
Comment (exceptionfactory) [inline nifi-nar-bundles/nifi-hadoop-bundle/nifi-hdfs-processors/src/test/java/org/apache/nifi/processors/hadoop/AbstractHadoopTest.java:144]: Should this be removed?

**Review by exceptionfactory — CHANGES_REQUESTED** [review]
Thanks for working through all of these modules @MikeThomsen!  Most of it looks good, but I noticed at least two issues with commented code and unchanged usage of JUnit 4 assertions.  Given the number of files changed, it might be helpful to break this into two or three separate pull requests.  I can take another look once you have addressed the noted items.

**Comment by MikeThomsen** [comment]
@exceptionfactory I'm going to close this and make several new smaller reviews out of its commits.

**Pull request closed without merging**